### PR TITLE
Task js dom

### PR DESCRIPTION
**Pull request by unabyband**
### task-JS-DOM

[Demo](https://unabyband.github.io/DOM-task/index.html) |
[Code base](https://github.com/unabyband/DOM-task)

Please, review.

### CODE REVIEW

**Comment by unabyband** [comment]
Hmmm, I didn't use loop for repaint, but used it for scanning events. Is this a bad practice too?

**Comment by unabyband** [comment]
Please, review

File: submissions/unabyband/task-js-dom/index.html [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    </head>
+    <body>
+        <header>
+            <a href="index.html"><div class="home-link"></div></a>
+            <span class="inner-text">Just a little visualisation about color temperature and white balance.</span>
+        </header>
+        <div class="main-container">
+            <nav class="sidebar">
+                <div class="side-menu inner-text" id="nav-menu">  
+                    <a href="#" class="side-menu-item" id="tab-0">Clear Sky</a>
```
Comment (lysenko-sergey-developer) [inline submissions/unabyband/task-js-dom/index.html:17]: Change `id` to` data-tab=`

File: submissions/unabyband/task-js-dom/script.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+//DOM functions here
+
+const mainContent = {
```
Comment (lysenko-sergey-developer) [inline submissions/unabyband/task-js-dom/script.js:3]: `mainContent` should store data not ids or html. Also, `mainContent` is an unclear name.

File: submissions/unabyband/task-js-dom/script.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+      "background-image: linear-gradient(lightblue, white, lightgray);",
+      "background-image: linear-gradient(lightgray, lightgreen, yellow);",
+      "background-image: linear-gradient(yellow, gold);",
+      "background-image: linear-gradient(gold, darkorange);",
+      "background-image: linear-gradient(orange, red);",
+    ],
+  };
+  
+  for (let i = 0; i < mainContent.sideTab.length; i++) {
+    document.querySelector(mainContent.sideTab[i]).onclick = function () {
```
Comment (lysenko-sergey-developer) [inline submissions/unabyband/task-js-dom/script.js:47]: With id by index, if we will add a new item to start or middle of the list, then we should change every idx after the appended element. Try to solve this issue.

File: submissions/unabyband/task-js-dom/script.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      document.querySelector(".content").innerHTML =
+        '<span class="inner-text">' +
+        mainContent.imgContent[i] +
+        "<br>" +
+        mainContent.textContent[i] +
+        "</span>";
+      document.querySelector(".content").style = mainContent.colorContent[i];
```
Comment (lysenko-sergey-developer) [inline submissions/unabyband/task-js-dom/script.js:54]: In this case better have classes with background colours.

**Comment by lysenko-sergey-developer** [comment]
@unabyband 👋 Good work!

**Review by lysenko-sergey-developer — CHANGES_REQUESTED** [review]

**Comment by lysenko-sergey-developer** [comment]
UI/UX Improvement:
Add margins/paddings for header/main-content/sidebar. When you have blocks with `border-radius` and this blocks stick to each other it looks bad.

File: submissions/unabyband/task-js-dom/index.html [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                    <a href="#" class="side-menu-item" id="tab-4">Fluorescent</a>
+                    <a href="#" class="side-menu-item" id="tab-5">Sunrise/Sunset</a>
+                    <a href="#" class="side-menu-item" id="tab-6">Domestic Light</a>
+                    <a href="#" class="side-menu-item" id="tab-7">Candle Flame</a>
+                </div>
+            </nav> 
+            <main class="content" id="content">
+                <span class="inner-text">
+                    White balance chart from different sources of light<br><br>
+                <img src="src/wb_chart.png" alt="white balance chart"></span>
```
Comment (lysenko-sergey-developer) [inline submissions/unabyband/task-js-dom/index.html:30]: `img` must have `alt` attribute.

File: submissions/unabyband/task-js-dom/index.html [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    <head>
+        <meta charset="UTF-8">
+        <meta name="viewport" content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
+        <link rel="stylesheet" href="css/main.css">
+        <title>unabyband DOM</title>
+    </head>
+    <body>
+        <header>
+            <a href="index.html"><div class="home-link"></div></a>
+            <span class="inner-text">Just a little visualisation about color temperature and white balance.</span>
```
[the remark this answers is not in the document]
Comment (unabyband) [reply submissions/unabyband/task-js-dom/index.html:12]: Done!

File: submissions/unabyband/task-js-dom/main.css [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+
+.side-menu-item:hover {
+    text-decoration: underline;
+    cursor: pointer;
+}
+
+.side-menu-item:focus-within {
+    background-color: skyblue;
+    border-radius: 10px;
+}
```
[the remark this answers is not in the document]
Comment (unabyband) [reply submissions/unabyband/task-js-dom/main.css:131]: Ok, thank you. Fixed.

File: submissions/unabyband/task-js-dom/index.html (hunk 1 above)
Comment (unabyband) [reply submissions/unabyband/task-js-dom/index.html:17]: Fixed. Sorry, but I just didn't know about "data attributes" in html at that time.

File: submissions/unabyband/task-js-dom/index.html (hunk 5 above)
Comment (unabyband) [reply submissions/unabyband/task-js-dom/index.html:30]: Definitely, and they all have "alt" attributes, even rendered from script file.

File: submissions/unabyband/task-js-dom/script.js (hunk 2 above)
Comment (unabyband) [reply submissions/unabyband/task-js-dom/script.js:3]: Done and done.

File: submissions/unabyband/task-js-dom/script.js (hunk 4 above)
Comment (unabyband) [reply submissions/unabyband/task-js-dom/script.js:54]: Wooow, it was not too simple, but - fixed!

File: submissions/unabyband/task-js-dom/script.js (hunk 3 above)
Comment (unabyband) [reply submissions/unabyband/task-js-dom/script.js:47]: Thanks for your notice. Fixed.
I remade this with "forEach", now it has to be better.

File: submissions/unabyband/task-js-dom/main.css [hunk 8]
```diff
@@ -0,0 +1,274 @@
...
+
+
+
+
+@media only screen and (max-width: 912px) {
```
[the remark this answers is not in the document]
Comment (unabyband) [reply submissions/unabyband/task-js-dom/main.css:136]: Finally, I did it. Woo-hoo

File: submissions/unabyband/task-js-dom/script.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+      ['src/lamps_light.jpg', 'Electric lamp light'],
+      ['src/candle_light.jpg', 'Candle light'],
+    ],
+    colorContent: ["blue", "skyblue", "lightblue", "white", "gray", "yellow", "gold", "orange"],
+  };
+  
+const selectors = document.querySelectorAll('.side-menu-item');
+let previousStyle = 'default';
+
+selectors.forEach((selector) => selector.onclick = function () {
```
Comment (lysenko-sergey-developer) [inline submissions/unabyband/task-js-dom/script.js:30]: Can we use event bubbling for use single event handler?

Comment (unabyband) [reply submissions/unabyband/task-js-dom/script.js:30]: Yes, but in that case we would need to check the selector data, and after that request appropriate content from our object.

In this case we just have eight selectors, each of them has it's own parameters in the "content" object. According to data attributes, they're rendering to our page.
This looks even simple and shorter in my opinion.

Comment (lysenko-sergey-developer) [reply submissions/unabyband/task-js-dom/script.js:30]: OK, let's skip this. But this is important to combine event handlers when it's possible.

**Review by lysenko-sergey-developer — APPROVED** [review]
LGTM! 😺

File: submissions/unabyband/task-js-dom/script.js (hunk 9 above)
Comment (unabyband) [reply submissions/unabyband/task-js-dom/script.js:30]: Ok, I get it! Thanks a lot for your review and advise. 
Today I worked at the friends-app and combined so many handlers in it 😎

**Pull request merged**